model/message_broker.go

```diff
@@ -0,0 +1,135 @@
+package model
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// MessageEntry represents an entry for a "message" of a topic in the database.
+// ForeignID is the id of a document located in another collection used by the
+// Topic interface to extract the actual message. The id of each message entry
+// is an int sequentially incremented for each message published to the topic.
+type MessageEntry struct {
+	ForeignID string `bson:"foreign_id"`
+
+	// ResumeToken and Error should be checked when watching a topic.
+	ResumeToken []byte
+	Err         error
+}
+
+// PublishToTopic creates a new message entry with the given id in the
+// corresponding collection for the given topic. If the topic does not exist in
+// the global registry, an error is returned.
+func PublishToTopic(ctx context.Context, env cedar.Environment, name string, id string) error {
```

Comment:
I guess it's premature since it isn't called anywhere, but how are we going to know when to publish to a topic?

Reply:
When people ask us to. We would have to create a new implementation of the `Topic` interface and then publish wherever the new information is coming in (probably in the rpc service code).

Reply:
Okay, I was just checking that every time we have to publish something for a new topic, we have to change the code. Instead of putting it in the gRPC code, isn't there a concern that this is fragile or should be put in the DB operation code? It basically means that anytime we do a database op, we'll have to check if the change is supposed to be published to some topic.

Reply:
Yeah it would be a code change, but hopefully with this interface not too hard or complicated. The `Topic` implementation would be put in the `model` packge and the `Publish` call would be in the `rpc` package (most likely, or maybe in an amboy job who knows).

Reply:
But if we are publishing something that may be updated or changed, then yes the call to `Publish` would probably make more sense in the database code.

Reply:
Alright, I'll leave it alone until I see what the publishing changes look like.